metrics.nim

```diff
@@ -231,6 +233,24 @@ template countExceptions*(counter: Counter, body: untyped) =
   counter.countExceptions(labelValues):
     body
 
+template counter*(identifier: untyped,
```

Comment:
This should still be named `newCounter`. I'm planning a secondary API where collectors are created on the fly, like Jacek envisioned, whenever necessary. It would look something like: `counter("metric_name_here").inc()`.

Reply:
I think the new API should be `incCounter "foo", 10`. Should be easy to do with a generic proc taking a `static string` counter name and creating a `.global.` variable.

Reply:
Think about gauges. We could have a single `gauge` proc that fetches or creates the corresponding Gauge object, or a multitude of procedures instead: `incGauge`, `decGauge`, `setGauge`, `setToCurrectTimeGauge`, `timeGauge`.

I think this is cleaner:
```nim
gauge("foo_bar").time:
    somethingSlow()
```

Reply:
OK, fair enough.